### ISSUE TITLE
Idf for BERTScore-style similarity computation?

### ISSUE THREAD

**Issue by jpowerj** [UKPLab/sentence-transformers#1333]
Hello,

For my dissertation I'm basically analyzing documents as sequences of (sBERT) sentence embeddings, then trying to compute *document* similarity scores in a manner analogous to standard STS methods.

I'm following the brilliant idea by @nreimers [here](https://github.com/UKPLab/sentence-transformers/issues/1166#issuecomment-918084006) of using [BERTScore](https://arxiv.org/abs/1904.09675) but with "token vector" replaced by "sBERT vector" so that the produced "sentence similarity" is actually a document similarity.

I just had a question on a detail, though: the final step in Figure 1:

![sbert](https://user-images.githubusercontent.com/2602912/146849872-5e20561f-e844-4020-85d0-fbbd5a0af2f4.png)

is to do importance weighting, by weighting the token-to-token similarities by an idf score. At first I thought to skip this step, but I'm wondering if actually I could compute an aggregate "sentence importance score" which is not the *sentence's* idf with respect to all sentences in the corpus (since most sentences probably only occur once ever) but rather the mean (or some other function) of its individual tokens' idf scores? This is to capture e.g. the notion that

> A = "Dies ist das Ende von Buch eins."

gives less semantic information about the document than e.g.

> B = "Dieses Buch präsentiert eine Kritik der kapitalistischen politischen Ökonomie."

and that this would be reflected in that the average idf score of the tokens in A would be lower than the average of the tokens in B (since e.g. `Ökonomie`, `kapitalistischen` occur less often than `Ende`, `Buch`, `eins`). Indeed, running it on my [quickly sloppily preprocessed for this] corpus, this gives idf(A) = 10.97 and idf(B) = 11.62, matching the intuition that B should be considered "more important" (for STS computation) than A.

So, I just wanted to make sure my intuition makes sense, that this wouldn't "break" the BERTScore analogy in a way I'm not realizing, since I've never used BERTScore until now.

tl;dr: Even though I'm only computing document vectors as BERTScore of [list of sentence vectors], can I "dip down" to the level of individual tokens just to get their average idf score?

[PS, your software is incredibly easy-to-use, and following your responses in this Issues section has helped me more times than I can count, so wanted to thank you for both! Plus it's a third bonus that UKPLab is in Germany and my whole dissertation corpus is German :D]

**Comment by nreimers** [comment]
Hi @jpowerj 
That is a nice idea. Yes, I think it is fine to compute the idf like this.
But as usual with machine learning: You do not really know. Hence, it make sense to test this on some type of evaluation corpus.

**Comment by jpowerj** [comment]
Thanks so so much, mostly I was looking for sanity check :D I will definitely test it on some known ground-truth data!

**Comment by naveenjafer** [comment]
@jpowerj I would love to know if you eventually used this method in your dissertation and if it bore fruit in terms of experimental results.
